From 683faa3aea411913275652fbc12d4fb172941bd6 Mon Sep 17 00:00:00 2001
From: Alex Bee <knaerzche@gmail.com>
Date: Mon, 1 Mar 2021 20:31:15 +0100
Subject: [PATCH 61/92] clk: rockchip: rk3288: use common PLL setting for 594
 MHz in NPLL table

The settings in the NPLL table (which were obviously copied from RK3368) don't
provide a stable signal for 594 MHz, what leads to random short-term black
screen periods (@2160p@60Hz) on some sensetive HDMI sinks when using this PLL
as the source for VOPs dclk.
Using the PLL settings from the common PLL table for this frequency fixes
this.
---
 drivers/clk/rockchip/clk-rk3288.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/rockchip/clk-rk3288.c b/drivers/clk/rockchip/clk-rk3288.c
index 233890555616..676e7c3c6f2b 100644
--- a/drivers/clk/rockchip/clk-rk3288.c
+++ b/drivers/clk/rockchip/clk-rk3288.c
@@ -122,7 +122,7 @@ static struct rockchip_pll_rate_table rk3288_pll_rates[] = {
 };
 
 static struct rockchip_pll_rate_table rk3288_npll_rates[] = {
-	RK3066_PLL_RATE_NB(594000000, 1, 99, 4, 32),
+	RK3066_PLL_RATE_NB(594000000, 1, 198, 8, 1),
 	RK3066_PLL_RATE_NB(585000000, 6, 585, 4, 32),
 	RK3066_PLL_RATE_NB(432000000, 3, 216, 4, 32),
 	RK3066_PLL_RATE_NB(426000000, 3, 213, 4, 32),
-- 
2.34.1

